File: bindings/kotlin/src/main/kotlin/at/schem/nucleation/GeneratedChunkCoverage.kt
package at.schem.nucleation

import com.sun.jna.Callback
import com.sun.jna.Library
import com.sun.jna.Native
import com.sun.jna.Pointer
import com.sun.jna.Structure

internal interface GeneratedChunkCoverageLib: Library {
}
/** Coverage of a generated chunk by its source graph.
*/
enum class GeneratedChunkCoverage {
    Complete,
    Partial,
    Outside;

    fun toNative(): Int {
        return this.ordinal
    }


    companion object {
        internal val libClass: Class<GeneratedChunkCoverageLib> = GeneratedChunkCoverageLib::class.java
        internal val lib: GeneratedChunkCoverageLib = Native.load("nucleation", libClass)
        fun fromNative(native: Int): GeneratedChunkCoverage {
            return GeneratedChunkCoverage.entries[native]
        }

        fun default(): GeneratedChunkCoverage {
            return Complete
        }
    }
}

File: bindings/php/GeneratedChunkCoverage.php
<?php
namespace Stencil;

final class GeneratedChunkCoverage {
    const Complete = 0;
    const Partial = 1;
    const Outside = 2;

    public static function name(int $value): string {
        return match ($value) {
            self::Complete => 'Complete',
            self::Partial => 'Partial',
            self::Outside => 'Outside',
            default => "Unknown({$value})",
        };
    }
}
